### DISCUSSION TITLE
Merge commits and PRs with one commit

### DISCUSSION THREAD

**Discussion by mgeraci** [community/community#7649] [Pull Requests] [answered]
Hi there! I have a small feature request having to do with pull requests that only have one commit. We use squash merging at OkCupid, and when I press "merge" on a PR with one commit, the commit message defaults to being the commit message of the merge commit, rather than the title of the pull request.

In my experience, this has been a small annoyance for me, as I often give better titles to pull requests than commit messages. You could definitely say I should name commits better! But I just ran into this: I had a commit called `use custom hook`, and a PR titled `Use custom hook to show reboarding app in onboarding`. And I would prefer the commit message in `main` to be `Use custom hook to show reboarding app in onboarding (#4480)` than `use custom hook (#4480)`.

Thanks for the consideration!

**Comment by blimmer** [comment]
100% agree on this. The behavior with squash & merge is inconsistent because it uses the PR title in some cases (> 1 commit) and the commit message in others. I understand that a single commit can be fast-forward merged, which is probably why the behavior is the way it is.

For my clients, I often use [semantic-release](https://github.com/semantic-release/semantic-release) to automate deployments. I have a GitHub action that validates that PR titles use [conventional-commits](https://www.conventionalcommits.org/en/v1.0.0/) syntax, which helps semantic-release figure out the proper next version number. However, this behavior breaks with single commit-message PRs because of this issue.

As a workaround, I'm using the [refined-github](https://github.com/refined-github/refined-github) browser extension, which "fixes" this behavior (see https://github.com/refined-github/refined-github/issues/276) by default.

**Comment by oliviertassinari** [comment]
Yes! Please change this! An illustration https://github.com/mui/material-ui/pull/30912#issuecomment-1032010495

**Comment by willsmythe** [comment, marked as answer]
The default commit message presented when merging is now configurable, and there's an option to use the PR title by default: https://github.blog/changelog/2022-08-23-new-options-for-controlling-the-default-commit-message-when-merging-a-pull-request/
